VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -297,6 +300,18 @@ def _post_to_wireserver_helper(self, msg_data, http_util):
                 else:
                     raise e
 
+    def get_fault_reason(self, content_xml):
+        try:
+            xml_root = ET.fromstring(content_xml)
+        except:
+            self.logger.log("Exception occured while parsing error xml.")
+            return "Unknown"
+        detail_element = xml_root.find('Details')
+        if detail_element is not None and (detail_element.text is not None and len(detail_element.text) > 0):
```

Comment:
could be potentially shrunk to:
"detail_element is not None and detail_element.text:"